lib/emque/producing/message/message.rb

```diff
@@ -97,21 +99,36 @@ def to_json
         Oj.dump(data, :mode => :compat)
       end
 
-      def publish(publisher=Emque::Producing.publisher)
+      def publish(publisher=nil)
+        publisher = Emque::Producing.publisher unless publisher
         log "publishing...", true
         if valid?
           log "valid...", true
           if Emque::Producing.configuration.publish_messages
-            sent = publisher.publish(topic, message_type, to_json, partition_key)
-            log "sent #{sent}"
-            if raise_on_failure? && !sent
-              raise MessagesNotSentError.new
+            begin
+              sent = publisher.publish(topic, message_type, to_json, partition_key)
+              log "sent #{sent}"
+              if raise_on_failure? && !sent
+                raise MessagesNotSentError.new
+              end
+            rescue *PUBLISH_IGNORED_EXCEPTIONS => error
+              if raise_on_failure?
+                raise
+              else
+                log "failed ignoring exception... #{error}", true
+              end
             end
           end
         else
           log "failed...", true
           raise InvalidMessageError.new(invalid_message)
         end
+      rescue *PUBLISH_IGNORED_EXCEPTIONS => error
```

Comment:
The inner one catches anything that happens during publisher.publish but I might be able to remove the inner rescue since the outer one should catch it as well.